catroid/src/main/java/org/catrobat/catroid/content/bricks/FormulaBrick.java

```diff
@@ -101,48 +147,35 @@ public void onClick(View view) {
 		showFormulaEditorToEditFormula(view);
 	}
 
+	public void showFormulaEditorToEditFormula(View view) {
+		if (brickFieldToTextViewIdMap.inverse().containsKey(view.getId())) {
+			FormulaEditorFragment.showFragment(view, this, brickFieldToTextViewIdMap.inverse().get(view.getId()));
+		} else {
+			FormulaEditorFragment.showFragment(view, this, formulaMap.keys().nextElement());
+		}
+	}
+
 	public View getCustomView(Context context, int brickId, BaseAdapter baseAdapter) {
 		return null;
 	}
 
-	public abstract void showFormulaEditorToEditFormula(View view);
-
-	public void setSecondText(View view, int textViewId, int editTextDurationId, BrickField brickField) {
-		TextView editDuration = view.findViewById(editTextDurationId);
-		getFormulaWithBrickField(brickField)
-				.setTextFieldId(editTextDurationId);
-		getFormulaWithBrickField(brickField).refreshTextField(view);
-
-		TextView times = view.findViewById(textViewId);
+	protected void setSecondsLabel(View view, BrickField brickField) {
+		TextView textView = view.findViewById(R.id.brick_seconds_label);
+		Context context = textView.getContext();
 
 		if (getFormulaWithBrickField(brickField).isSingleNumberFormula()) {
 			try {
-				times.setText(view.getResources().getQuantityString(
-						R.plurals.second_plural,
-						Utils.convertDoubleToPluralInteger(getFormulaWithBrickField(brickField)
-								.interpretDouble(ProjectManager.getInstance().getCurrentSprite()))
-				));
-			} catch (InterpretationException interpretationException) {
-				Log.d(getClass().getSimpleName(), "Couldn't interpret Formula.", interpretationException);
+				Double formulaValue = formulaMap.get(brickField)
+						.interpretDouble(ProjectManager.getInstance().getCurrentSprite());
+				textView.setText(context.getResources().getQuantityString(R.plurals.second_plural,
+						Utils.convertDoubleToPluralInteger(formulaValue)));
+			} catch (InterpretationException e) {
+				Log.e(getClass().getSimpleName(),
+						"Interpretation of formula failed, fallback to quantity \"other\"", e);
 			}
-		} else {
-			times.setText(view.getResources().getQuantityString(R.plurals.second_plural,
-					Utils.TRANSLATION_PLURAL_OTHER_INTEGER));
+			return;
 		}
-
-		editDuration.setOnClickListener(this);
-	}
-
-	public void setSecondText(Context context, View prototypeView, int textViewId) {
-		TextView second = prototypeView.findViewById(textViewId);
-		second.setText(context.getResources().getQuantityString(R.plurals.second_plural,
-				Utils.convertDoubleToPluralInteger(BrickValues.DRONE_MOVE_BRICK_DEFAULT_TIME_MILLISECONDS / 1000)));
-	}
-
-	protected <T> String formatNumberForPrototypeView(T value) {
-
-		String number = String.valueOf(value);
-
-		return FormatNumberUtil.cutTrailingZeros(number);
+		textView.setText(context.getResources()
+				.getQuantityString(R.plurals.second_plural, Utils.TRANSLATION_PLURAL_OTHER_INTEGER));
```

Comment:
Maybe if-else instead?

Reply:
well, maybe.that depends on your beliefs I guess, but sure, I can change it.

Reply:
No, wait!

I guess the return is wrong here https://github.com/thmq/Catroid/blob/86ebc59f657cd91334a8476a18a44af3633ce2bd/catroid/src/main/java/org/catrobat/catroid/content/bricks/FormulaBrick.java#L176

Reply:
Or at least that's what the log message says.

Reply:
oh, yes! it should fallback to "other" -> :scream: I messed up something there.

Reply:
Kudos to your improved log message! :star:

Reply:
thank you :kissing_closed_eyes:

Reply:
so, found the issue: `return` should have been in the `try` block -> so that if an exception is thrown or the formula isn't a single number it falls back to "other".

however, I think this should be changed anyway as we cannot assume that just because it is correct to say `someVariableValue seconds` opposed to `someVariableValue second` in English it is also correct to say that in other languages.